arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,867 @@
+        }
+      }
+    },
+    "OperationProperties": {
+      "description": "Properties of an operation supported by the resource provider.",
+      "type": "object",
+      "properties": {
+        "description": {
+          "description": "The description of the resource operation.",
+          "type": "string"
+        },
+        "operation": {
+          "description": "The operation name.",
+          "type": "string"
+        },
+        "provider": {
+          "description": "The provider name.",
+          "type": "string"
+        },
+        "resource": {
+          "description": "The resource name.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceGetResponse": {
+      "description": "The response to a resource GET request.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique identifier of the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "description": "Resource properties.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ResourceGroupGetResponse": {
+      "description": "The response to a resource group GET request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of resource details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceGetResponse"
+          }
+        }
+      }
+    },
+    "ResourceOperationsDescriptor": {
+      "description": "Properties of an operation supported by the resource provider.",
+      "type": "object",
+      "properties": {
+        "display": {
+          "description": "The properties of the resource operation.",
+          "$ref": "#/definitions/OperationProperties"
+        },
+        "name": {
+          "description": "The name of the resource operation.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceOperationsDescriptorWrapper": {
+      "description": "Container for a list of operations supported by a resource provider.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "A list of operations supported by a resource provider.",
+          "type": "array",
```

Comment:
Add `"readOnly": true` attribute so that the generated SDKs won't expose public setters for the collection.